aws/cloudformation/iam.yml.erb

```diff
@@ -835,6 +835,90 @@ roles.each do |name, config| %>
     Type: AWS::IAM::ServiceLinkedRole
     Properties: 
       AWSServiceName: logger.cloudfront.amazonaws.com
+
+  # ------------------------------------------------------------
+  # Create one AWS IAM User for each type of marketing sites environment that GitHub Actions will create, with
+  # permissions needed to create/update/delete a marketing sites CloudFormation Stack.
+  # ------------------------------------------------------------
+  <%
+    marketing_sites_environment_type_configuration = {
+      development: {
+        base_domain_name: 'dev-code.org'
+      },
+      test: {
+        base_domain_name: 'test-code.org'
+      },
+      production: {
+        base_domain_name: 'code.org'
+      }
+    }
+    marketing_sites_environment_type_configuration.each do |environment_type, configuration|
```

Comment:
if these have specific domain names in their config, then thisis an `environment_name` not an `environment_type`

Reply:
We'd also need to differentiate between clusters if we are intending to have different clusters

Reply:
> if these have specific domain names in their config, then this is an `environment_name` not an `environment_type`

Each environment type is associated with a *base* domain name, not a fully qualified domain name. All Stacks of the same environment type use the same base domain name.

Reply:
> I agree with your previous statement that `environment` is an overloaded term.
> 
> Since we need to differentiate between clusters if we are intending to have different clusters, perhaps we can have a term that expresses the usage of each stack? Perhaps `marketing_stack_name`?

`cluster` is an overloaded term as well since we’re using ECS Fargate 😅

This proposal is to create 3 Roles; one for each of the 3 environment types (`development`, `test`, and `production`). I’m intentionally holding off on provisioning more fine-grained controls (one deployer Role per Stack) under the assumption that the risks of a test Hour of Code deployment making changes to a test Teach AI Stack.

Reply:
:+1: